src/node_url.cc

```diff
@@ -1477,6 +1477,8 @@ void URL::Parse(const char* input,
           if (ch != '/') {
             continue;
           }
+        } else if (has_state_override && url->host.empty()) {
```

Comment:
@addaleax The fifth step of [path start state](https://url.spec.whatwg.org/#path-start-state) which I tried to implement here says:

> Otherwise, if state override is given and url’s host is null, append the empty string to url’s path.

Shouldn't it say that the url's host is *empty* instead of *null* or should we refactor our implementation to change the type of `url_data::host` to a char pointer initially set to `nullptr` as mentioned in [here](https://url.spec.whatwg.org/#concept-url-host)?

Reply:
@RaisinTen Maybe? If there is a difference, shouldn’t the spec tests cover it?

Reply:
@addaleax The specs have landed but the corresponding wpt tests PR didn't: https://github.com/web-platform-tests/wpt/pull/27720. :/

Reply:
@RaisinTen I think in that case we should try to get that landed soon :)

Reply:
I got the tests landed.